xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -1240,12 +1240,21 @@ public String getRenderedContent(Syntax targetSyntax, boolean transformationCont
         // rights
         Object currrentSdoc = context.get("sdoc");
         try {
-            // If we execute a translation use translated document as secure document
-            XWikiDocument tdoc = getTranslatedDocument(context);
+            XWikiDocument sdoc;
+            boolean translate;
 
-            context.put("sdoc", tdoc);
+            if (this.locale != Locale.ROOT) {
```

Comment:
The locale is always set in a XWikiDocument coming from the DB.

I really don't think there is any criteria you can check for that `if`, the only way is fix this issue IMO is a new method which explicitly take the locale as parameter when you don't want to use the context one.

Reply:
> The locale is always set in a XWikiDocument coming from the DB.

So it can also be set to ROOT right? 
This change is working on my integration test with a doc with 2 translations.

Reply:
> So it can also be set to ROOT right?

My point is that it's most use cases it's set to ROOT since we generally use this method from the ROOT XWikiDocument instance.